docs/RFCS/20171118_role_based_access_control.md

```diff
@@ -0,0 +1,757 @@
+of the corresponding roles. The alternative would be granting privileges for all necessary objects,
+making it likely to forget some.
+
+Managing privileges on a given object becomes easier: by changing the privileges granted to the role,
+all members of the role are updated implicitly.
+
+# Related resources
+
+We follow the postgresql use of roles:
+* [PostgreSQL user management](https://www.postgresql.org/docs/current/static/user-manag.html)
+* [PostgreSQL CREATE ROLE](https://www.postgresql.org/docs/10/static/sql-createrole.html)
+* [PostgreSQL DROP ROLE](https://www.postgresql.org/docs/10/static/sql-droprole.html)
+* [PostgreSQL GRANT on ROLES](https://www.postgresql.org/docs/current/static/sql-grant.html#SQL-GRANT-DESCRIPTION-ROLES)
+* [PostgreSQL REVOKE](https://www.postgresql.org/docs/current/static/sql-revoke.html)
+
+# Out of scope
+
+The following are not in scope but should not be hindered by implementation of this RFC:
+* role attributes (see [PostgreSQL CREATE ROLE](https://www.postgresql.org/docs/10/static/sql-createrole.html) for full list)
+* role auditing
+* multiple "admin" (or superuser) roles
+* admin UI manipulation of roles
+* new privileges (eg: admin UI related)
+
+See [Future improvements](#future-improvements) for discussion of some of these.
+
+# Guide-level explanation
+
+## Terminology
+
+Some terminology used in this RFC:
+* **role**: the focus point of this RFC. Please re-read the [summary](#summary)
+* **role admin**: a member of the role allowed to change role membership
+* **admin**: the special role allowed superuser operations, always exists
+* **root**: the special user existing by default as a member of the `admin` role
+* **inherit**: the behavior that grants a role's privileges to its members
+* **`A ∈ B`**: user or role `A` is a member of role `B`
+* **direct member**: A is a direct member of B if `A ∈ B`
+* **indirect member**: A is an indirect member of B if `A ∈ C ... ∈ B` where `...` is be an arbitrary number of memberships
+
+## User-level explanation
+
+### Role basics
+
+A role can be thought of as a group with any number of members belonging to the group.
+Roles have privileges on objects in the same way that users do. All members of the role
+inherit the privileges held by the role. This is done recursively.
+
+For example:
+* role `employees` has `ALL` privileges on table `mydb.employee_data`
+* user `marc` is a member of `employees` (also written: `marc ∈ employees`
+* without any extra privileges, `marc` can now perform any operations on table `mydb.employee_data`
+
+Roles and users follow a small set of rules:
+* roles and users use the same namespace (ie: if a user `marc` exists, we cannot create a role `marc`)
+* users and roles can be members of roles
+* membership loops are not allowed (direct: `A ∈ B ∈ A` or indirect: `A ∈ B ∈ C ... ∈ A`)
+* all privileges of a role are inherited by all its members
```

Comment:
A member cannot override the privileges granted to it from its parent role by revoking the privilege

Reply:
I don't know what that means. A given user has exactly the privileges granted to itself and all the roles it's a member of (directly or indirectly).